compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -3010,6 +3010,7 @@ declare_lint_pass! {
         UNSUPPORTED_CALLING_CONVENTIONS,
         BREAK_WITH_LABEL_AND_LOOP,
         UNUSED_ATTRIBUTES,
+        NON_EXHAUSTIVE_REACHABLE_PATTERNS,
```

Comment:
The lint name isn't clear. `non_exhaustive_reachable_wildcards` would be an improvement, but maybe we can do even better